test/session_test.exs

```diff
@@ -39,5 +39,16 @@ defmodule Janus.SessionTest do
 
       assert_receive {:trace, ^conn, :receive, %{"janus" => "ack"}}, 2 * interval
     end
+
+    test "stop on connection exit", %{connection: conn} do
+      Process.flag(:trap_exit, true)
+
+      {:ok, session} = Session.start_link(conn, @timeout)
```

Comment:
You should probably provide a way to start this process without link (usually, `start` and `start_link` look the same except for the last call, so you can handle it with one private function like in `Membrane.Pipeline`: https://github.com/membraneframework/membrane_core/blob/19c98a8bb24b30ff2f116f63c299cd66570ba4aa/lib/membrane/pipeline.ex#L193)

Reply:
Besides providing `start` function, should the test take advantage of it in any way?

Reply:
Just that the test should not `trap_exit`